File: pyg_parent/pyg_manager_web/src/main/webapp/js/controller/brandController.js
//定义控制器
app.controller("brandController",function ($scope,$controller,brandService) {
    // //读取列表数据绑定到表单中
    // $scope.findAll=function () {
    // $http.get("../brand/findAll.do").success(function (response) {
    // 	$scope.list=response;
    //     });
    // }

    $controller("baseController",{$scope:$scope});//继承

    //分页
    $scope.findPage=function(page,size) {
        brandService.findPage(page,size).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    };


    //保存
    $scope.save=function () {
        var object=null;
        if($scope.entity.id!=null){
            object=brandService.update($scope.entity);
        }else {
            object=brandService.add($scope.entity);
        }

        object.success(function (response) {
            if(response.success){
                // alert("dd");
                $scope.reloadList();//刷新
                // alert("ee");
            }else {
                alert(response.message)
            }
        })
    };

    //查询实体

    $scope.findOne=function (id) {

        //alert(id);
        brandService.findOne(id).success(function (response) {
            $scope.entity=response;
            //alert($scope.entity);
        })
    };

    //删除
    $scope.dele=function () {
        //获取选中的复选框
        brandService.dele($scope.selectIds).success(function (response) {
            if(response.success){
                //alert("aaa");
                $scope.reloadList();
                //alert("bbb");
            }else {
                //alert("ccc");
                alert(response.message)
            }
        })
    };
//条件查询
    $scope.searchEntity={};
    $scope.search=function (page,size) {
        brandService.search(page,size,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    };


});

File: pyg_parent/pyg_manager_web/src/main/java/com/pyg/web/BrandController.java
package com.pyg.web;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.pojo.TbBrand;
import com.pyg.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    //从远程获取接口的服务实现
    @Reference
    private BrandService brandService;

    @RequestMapping("findAll")
    public List<TbBrand> findAll(){
       return  brandService.findAll();
    }

    @RequestMapping("findPage")
    public PageResult findPage(int page,int size){
        return brandService.findPage(page,size);
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }
    @RequestMapping("findOne")
    public TbBrand findOne(long id){
       return brandService.findOne(id);
    }

    @RequestMapping("update")
    public Result update(@RequestBody TbBrand brand){
       // System.out.println("aaa");
        try {
            brandService.update(brand);
           // System.out.println("bbb");
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    @RequestMapping("delete")
    public Result delete(long[] ids){
        try {
            brandService.delete(ids);
            return  new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    @RequestMapping("search")
    public PageResult search(@RequestBody TbBrand brand,int page,int size){
        //System.out.println("aaa");
        return brandService.search(brand, page, size);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }

}
